sdk/trace/span.go

```diff
@@ -225,9 +224,17 @@ func (s *span) End(options ...trace.SpanEndOption) {
 		return
 	}
 
+	// If the span was not properly started, stopwatch will be nil.
+	// The normal check (IsRecording) happens after end time generation,
+	// so we add a simple check here to make sure end time is generated
+	// as early as possible.
+	if s.stopwatch == nil {
+		return
+	}
+
 	// Store the end time as soon as possible to avoid artificially increasing
 	// the span's duration in case some operation below takes a while.
-	et := internal.MonotonicEndTime(s.startTime)
+	et := s.stopwatch.Started().Add(s.stopwatch.Stop())
```

Comment:
what if `s.stopwatch.Stop()` returns a negative value?

I guess that if we really want to ensure monotonicity, then probably we should trim it to `0` in such case

Reply:
Was going to post about this. I think a check for negative duration is reasonable but I'd like to gather more input about this on whether we should really do it. BTW if a check is added I would also need to adjust one of my test case (which purposely returns a negative duration just to test the functionality).

Reply:
I would avoid explicitly adding a guard for negative duration. I do not have a good example of when a user will want to do this (maybe a negative duration span means something to them(?)) but it would only remove that ability from a custom implementation. I think as long as we guarantee the default implementation does not do this it will be sufficient.

Reply:
_As for me, we can resolve this conversation_